exo-deploy/terraform/aws/alb-security-groups/security-groups.tf

```diff
@@ -0,0 +1,64 @@
+resource "aws_security_group" "internal_alb" {
+  name        = "${var.name}-internal-alb"
+  vpc_id      = "${var.vpc_id}"
+  description = "Allows internal ALB traffic"
+
+  ingress {
```

Comment:
If you don't, then you can't add rule to the security group after the fact (both methods are incompatible). Adding rules outside the defining module is sometimes necessary. Since it's functionally equivalent, the more flexible option is preferable. See [docs](https://www.terraform.io/docs/providers/aws/r/security_group_rule.html)